services/brig/src/Brig/Provider/RPC.hs

```diff
@@ -60,10 +64,23 @@ data ServiceError
 -- or the response body cannot be parsed, a 'ServiceError' is returned.
 createBot :: ServiceConn -> NewBotRequest -> ExceptT ServiceError AppIO NewBotResponse
 createBot scon new = do
-    mg <- ($ toList (sconFingerprints scon)) <$> view extGetManager
+    let fprs = toList (sconFingerprints scon)
+    (man, verifyFingerprints) <- view extGetManager
     extHandleAll onExc $ do
-        rs <- lift $ recovering x3 httpHandlers $ const $
-            liftIO $ Http.httpLbs req mg
+        rs <- lift $ recovering x3 httpHandlers $ const $ liftIO $
+            -- TODO: not sure if 'Reuse' is the right thing to do here --
+            -- maybe instead of returning the connection to the pool we want
+            -- to close it
+            Http.withConnection' req man Http.Reuse $ \mConn -> do
+                -- If we see this connection for the first time, verify fingerprints
+                let conn = Http.managedResource mConn
+                    seen = Http.managedReused   mConn
+                case (seen, fromDynamic @SSL.SSL (Http.connectionRaw conn)) of
+                    (True, _) -> pure ()
+                    (_, Nothing) -> error "TODO"
```

Comment:
So this should not be possible as services [only allow SSL connections](https://github.com/wireapp/wire-server/blob/develop/libs/brig-types/src/Brig/Types/Provider.hs#L430): Maybe we can fail in this case?

Reply:
I'd rather keep the `error` here if it's indeed not possible. I changed the error message though.